ScoutSuite/__main__.py

```diff
@@ -0,0 +1,210 @@
+            for service in cloud_provider.service_list:
+                cloud_provider.services[service] = current_run_services[service]
+
+    # Partial run, using pre-pulled data
+    else:
+        print_info('Using local data')
+        # Reload to flatten everything into a python dictionary
+        last_run_dict = report.encoder.load_from_file(ReportFile.results)
+        for key in last_run_dict:
+            setattr(cloud_provider, key, last_run_dict[key])
+
+    # Pre processing
+    cloud_provider.preprocessing(
+        args.get('ip_ranges'), args.get('ip_ranges_name_key'))
+
+    # Analyze config
+    print_info('Running rule engine')
+    finding_rules = Ruleset(environment_name=args.get('profile'),
+                            cloud_provider=args.get('provider'),
+                            filename=args.get('ruleset'),
+                            ip_ranges=args.get('ip_ranges'),
+                            aws_account_id=cloud_provider.aws_account_id)
+    processing_engine = ProcessingEngine(finding_rules)
+    processing_engine.run(cloud_provider)
+
+    # Create display filters
+    print_info('Applying display filters')
+    filter_rules = Ruleset(cloud_provider=args.get('provider'),
+                           filename='filters.json',
+                           rule_type='filters',
+                           aws_account_id=cloud_provider.aws_account_id)
+    processing_engine = ProcessingEngine(filter_rules)
+    processing_engine.run(cloud_provider)
+
+    if args.get('exceptions')[0]:
+        print_info('Applying exceptions')
+        try:
+            exceptions = RuleExceptions(
+                args.get('profile'), args.get('exceptions')[0])
+            exceptions.process(cloud_provider)
+            exceptions = exceptions.exceptions
+        except Exception as e:
+            print_debug(
+                'Failed to load exceptions. The file may not exist or may have an invalid format.')
+            exceptions = {}
+    else:
+        exceptions = {}
+    # Handle exceptions
+    try:
+        exceptions = RuleExceptions(
+            args.get('profile'), args.get('exceptions')[0])
+        exceptions.process(cloud_provider)
+        exceptions = exceptions.exceptions
+    except Exception as e:
+        print_debug(
+            'Warning, failed to load exceptions. The file may not exist or may have an invalid format.')
+        exceptions = {}
+
+    # Finalize
+    cloud_provider.postprocessing(report.current_time, finding_rules)
+
+    # Save config and create HTML report
+    html_report_path = report.save(cloud_provider, exceptions, args.get('force_write'),
+                                   args.get('debug'))
+
+    # Open the report by default
+    if not args.get('no_browser'):
+        print_info('Opening the HTML report')
+        url = 'file://%s' % os.path.abspath(html_report_path)
+        webbrowser.open(url, new=2)
+
+    return 0
+
+
+def generate_report_name(provider_code, args):
```

Comment:
Can we move this to a util file? I feel like it doesn't belong to the `__main__.py`.

Reply:
This has already been removed in `develop`, merging this branch will be **fun**.

Reply:
Ugh yeah, I'm trying to merge develop into this in a separate branch to not break the testing of the functionality here. But the fact that : I changed all the lines in .js files and changed all the `scout2` s to `scoutsuite` while @j4v changed them to `scout`, on develop `aws_account_id` was changed to `account_id` which is a very small deal, but since I changed everything in scoutsuite.js it is a slighty larger deal as the IDE doesn't highlight the merge conflicts properly anymore 😂,  the stuff moved from `__main__` to `scout.py` on develop while new stuff was added on our side in __main__, just to name a few....